src/sinks/aws_cloudwatch_logs/config.rs

```diff
@@ -85,20 +90,46 @@ impl CloudwatchLogsSinkConfig {
         )
         .await
     }
+
+    pub async fn create_smithy_client(
+        &self,
+        proxy: &ProxyConfig,
+    ) -> crate::Result<aws_smithy_client::Client> {
+        let region = match self.region.region() {
```

Comment:
It's probably worth breaking this out into a function in `src/aws/mod.rs` and calling it instead.